src/test/java/unit/GcsBucketReferenced.java

```diff
@@ -27,26 +30,42 @@
 public class GcsBucketReferenced extends SingleWorkspaceUnit {
 
   // external bucket to use for creating GCS bucket references in the workspace
-  private BucketInfo externalBucket;
+  private BucketInfo externalSharedBucket;
+  private BucketInfo externalPrivateBucket;
+
+  private TestUsers shareeUser;
 
   @BeforeAll
   @Override
   protected void setupOnce() throws Exception {
     super.setupOnce();
-    externalBucket = ExternalGCSBuckets.createBucketWithUniformAccess();
-    ExternalGCSBuckets.grantReadAccess(externalBucket, Identity.user(workspaceCreator.email));
+    externalSharedBucket = ExternalGCSBuckets.createBucketWithUniformAccess();
+    externalPrivateBucket = ExternalGCSBuckets.createBucketWithUniformAccess();
+
+    ExternalGCSBuckets.grantWriteAccess(
+        externalSharedBucket, Identity.user(workspaceCreator.email));
+    ExternalGCSBuckets.grantReadAccess(
+        externalPrivateBucket, Identity.user(workspaceCreator.email));
```

Comment:
Since you are granting access to the proxy group below, I don't think you need to grant permission to the user's email, since that should be included in their proxy group.